### PR TITLE
Apple TV support 2: Xcode projects and CI (scripts/objc-test.sh)

### PR DESCRIPTION
**Pull request by dlowder-salesforce**
* Motivation *

Second PR for Apple TV support.

* Test plan *

Apple TV tests have been added to scripts/objc-test.sh

### CODE REVIEW

**Comment by dlowder-salesforce** [comment]
Hmmm... Travis CI failed because their build doesn't have a tvOS 10 simulator.  Will downgrade the build targets and see if that fixes it.

**Comment by dlowder-salesforce** [comment]
@javache here is Apple TV round 2 :)

File: React/Views/RCTScrollView.m [hunk 1]
```diff
@@ -946,6 +955,9 @@ - (type)getter                                            \
 #if !TARGET_OS_TV
 RCT_SET_AND_PRESERVE_OFFSET(setPagingEnabled, isPagingEnabled, BOOL)
 RCT_SET_AND_PRESERVE_OFFSET(setScrollsToTop, scrollsToTop, BOOL)
+#else
+RCT_EMPTY_BOOL_GETTER_SETTER(setPagingEnabled, isPagingEnabled)
```
Comment (javache) [inline React/Views/RCTScrollView.m]: Why is this necessary? I believe RN wil automatically ignore attributes that are not exported by a view manager (which is how we OS-specific attributes)

File: packager/react-native-xcode.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ case "$CONFIGURATION" in
   Debug)
     # Speed up build times by skipping the creation of the offline package for debug
     # builds on the simulator since the packager is supposed to be running anyways.
-    if [[ "$PLATFORM_NAME" = "iphonesimulator" ]]; then
+    if [[ "$PLATFORM_NAME" = "iphonesimulator" || "$PLATFORM_NAME" = "appletvsimulator" ]]; then
```
Comment (javache) [inline packager/react-native-xcode.sh]: Maybe regex match on 'simulator$'?

File: React/Views/RCTScrollView.m (hunk 1 above)
Comment (dlowder-salesforce) [inline React/Views/RCTScrollView.m]: Thanks to your hint I found the correct way to do this... committing a fix.

File: packager/react-native-xcode.sh (hunk 2 above)
Comment (dlowder-salesforce) [inline packager/react-native-xcode.sh]: Done.

File: Examples/UIExplorer/UIExplorerUnitTests/RCTBundleURLProviderTests.m [hunk 3]
```diff
@@ -73,6 +73,7 @@ - (void)tearDown
 
 - (void)testBundleURL
 {
+#if !TARGET_OS_TV
```
Comment (javache) [inline Examples/UIExplorer/UIExplorerUnitTests/RCTBundleURLProviderTests.m]: How is this this test broken on tvOS?

File: React/Modules/RCTUIManager.h [hunk 4]
```diff
@@ -97,7 +97,6 @@ RCT_EXTERN NSString *const RCTUIManagerRootViewKey;
  * @param reactTag the component tag
  * @param completion the completion block that will hand over the rootView, if any.
  *
- * @return the rootView
```
Comment (javache) [inline React/Modules/RCTUIManager.h]: Revert?

File: React/React.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -79,6 +79,90 @@
...
+		2D3B5EEC1D9B09D400451313 /* RCTTabBarItemManager.m in Sources */ = {isa = PBXBuildFile; fileRef = 137327E41AA5CF210034F82E /* RCTTabBarItemManager.m */; };
+		2D3B5EED1D9B09D700451313 /* RCTTabBarManager.m in Sources */ = {isa = PBXBuildFile; fileRef = 137327E61AA5CF210034F82E /* RCTTabBarManager.m */; };
+		2D3B5EEE1D9B09DA00451313 /* RCTView.m in Sources */ = {isa = PBXBuildFile; fileRef = 13E067501A70F44B002CDEE1 /* RCTView.m */; };
+		2D3B5EEF1D9B09DC00451313 /* RCTViewManager.m in Sources */ = {isa = PBXBuildFile; fileRef = 13E0674E1A70F44B002CDEE1 /* RCTViewManager.m */; };
+		2D3B5EF01D9B09E300451313 /* RCTWrapperViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = 13B080241A694A8400A75B9A /* RCTWrapperViewController.m */; };
+		2D3B5EF11D9B09E700451313 /* UIView+React.m in Sources */ = {isa = PBXBuildFile; fileRef = 13E067541A70F44B002CDEE1 /* UIView+React.m */; };
+		2D8C2E331DA40441000EE098 /* RCTMultipartStreamReader.m in Sources */ = {isa = PBXBuildFile; fileRef = 001BFCCF1D8381DE008E587E /* RCTMultipartStreamReader.m */; };
```
Comment (javache) [inline React/React.xcodeproj/project.pbxproj]: Any idea why all of these were added?

File: React/Views/RCTModalHostViewController.m [hunk 6]
```diff
@@ -73,5 +72,7 @@ - (UIInterfaceOrientationMask)supportedInterfaceOrientations
   return _supportedInterfaceOrientations;
 }
 #endif
+#endif
```
Comment (javache) [inline React/Views/RCTModalHostViewController.m]: Can you add a comment here to clarify what each of the if's are closing?

File: React/Views/RCTScrollView.m [hunk 7]
```diff
@@ -930,7 +930,7 @@ - (type)getter                                            \
 {                                                         \
   return [_scrollView getter];                            \
 }
-
+
```
Comment (javache) [inline React/Views/RCTScrollView.m]: Revert :)

File: React/Modules/RCTUIManager.h (hunk 4 above)
Comment (dlowder-salesforce) [inline React/Modules/RCTUIManager.h]: That change fixes an actual doc error -- the method does not in fact return anything.

File: React/React.xcodeproj/project.pbxproj (hunk 5 above)
Comment (dlowder-salesforce) [inline React/React.xcodeproj/project.pbxproj]: Those are to include the files in the tvOS build.

File: React/Views/RCTModalHostViewController.m (hunk 6 above)
Comment (dlowder-salesforce) [inline React/Views/RCTModalHostViewController.m]: Done.

File: React/Views/RCTScrollView.m (hunk 7 above)
Comment (dlowder-salesforce) [inline React/Views/RCTScrollView.m]: Done.

**Comment by javache** [comment]
@facebook-github-bot shipit

**Comment by bestander** [comment]
@dlowder-salesforce, I think this PR's unit tests are consistently failing:
```
Executed 123 tests, with 0 failures (0 unexpected) in 5.127 (5.227) seconds
All tests
Test Suite UIExplorer-tvOSIntegrationTests.xctest started
RCTLoggingTests
    ✗ testLogging, ((_lastLogLevel) equal to (RCTLogLevelError)) failed: ("2") is not equal to ("3")
    ✗ testLogging, ((_lastLogMessage) equal to (@"Invoking console.error")) failed: ("Warning: Native component for "RCTDatePicker" does not exist") is not equal to ("Invoking console.error")
    ✗ testLogging,
RCTRootViewIntegrationTests
    ✗ testPropertiesUpdateTest, failed: caught "NSInternalInconsistencyException", "RedBox error: Unhandled JS Exception: undefined is not an object (evaluating 'StatusBarManager.HEIGHT')"
    ✗ testPropertiesUpdateTest,
    ✗ testReactContentSizeUpdateTest, failed: caught "NSInternalInconsistencyException", "RedBox error: Unhandled JS Exception: Module AppRegistry is not a registered callable module (calling runApplication)"
```
https://travis-ci.org/facebook/react-native/jobs/165251326

**Comment by douglowder** [comment]
Weird.... the tests were passing locally.

**Comment by bestander** [comment]
It could be sensitive to CI device performance or XCode version

**Comment by douglowder** [comment]
Hmmm maybe put the Apple TV objc-test.sh in a separate script?

**Comment by bestander** [comment]
Yeah, let's give it a try.
Either this or disable it for now, otherwise we lose signal from all iOS
builds.
